core/src/main/java/org/apache/iceberg/io/FanoutWriter.java

```diff
@@ -55,7 +55,7 @@ public void write(T row, PartitionSpec spec, StructLike partition) {
   }
 
   private FileWriter<T, R> writer(PartitionSpec spec, StructLike partition) {
-    Map<StructLike, FileWriter<T, R>> specWriters = writers.computeIfAbsent(
+    StructLikeMap<FileWriter<T, R>> specWriters = writers.computeIfAbsent(
```

Comment:
:+1: I think this was fine before just because the actual implementation that uses this always returns a StructLikeMap (See line 60).

Reply:
reverted

Reply:
I don't think you should revert your change here. I just was checking to make sure it wasn't bugged